src/extension/uriRebaser.ts

```diff
@@ -23,12 +23,43 @@ function splitUri(uri: string | undefined) {
     return [`${scheme}://${authority}`, ...path.slice(1).split('/')]; // By spec first '/' always exists, thus safe to slice(1).
 }
 
-// Returns the distinct file name, instead of `true`, becuase in some cases the file name is used.
+const workspaceDistinctFilenameCache: Map<string, string | undefined> = new Map();
+
 async function workspaceHasDistinctFilename(filename: string): Promise<string | undefined> {
+    if (workspaceDistinctFilenameCache.has(filename)) {
+        return workspaceDistinctFilenameCache.get(filename);
+    }
     const matches = await workspace.findFiles(`**/${filename}`); // Is `.git` folder excluded?
-    return matches.length === 1 ? matches[0].toString() : undefined;
+    const result = matches.length === 1 ? matches[0].toString() : undefined;
+
+    workspaceDistinctFilenameCache.set(filename, result);
+    return result;
 }
 
+workspace.onDidCreateFiles(async (event) => {
+    for (const file of event.files) {
+        const filename = path.basename(file.path);
+        workspaceDistinctFilenameCache.delete(filename);
```

Comment:
>delete [](http://example.com/codeflow?start=39&length=6)

Why isn't this a set?

Reply:
Simplifying the logic by only doing cache invalidation, and leaving cache population to `workspaceHasDistinctFilename`.

Avoids the need for explicit edge case handling. For one consider if 'filename' not found in cache. It is not possible to tell if it is because no previous file exist, or if it is simply the cache hasn't been setup.

Reply:
Gotcha.